xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -212,7 +213,18 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Exclude the first heading from the content
+        if (parameters.excludeFirstHeading()) {
+            HeaderBlock heading =
+                    result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class), Block.Axes.DESCENDANT_OR_SELF);
+            if (heading != null) {
+                result.removeBlock(heading);
+            } else {
+                throw new MacroExecutionException("The included document or section doesn't contain any heading");
```

Comment:
Actually this message could be improved a bit since we know if the user is including a section or not.

Reply:
I tried generating this message dynamically using this approach:
```
                String includeType = parameters.getSection() != null ? "section" : "document";
                throw new MacroExecutionException("The included " + includeType + " doesn't contain any heading"); 
```
But I got mojojo check: `Coverage checks have not been met.` error. And If I use simple if else it builds successfully.

Reply:
I think the instructions not covered completely and due to some test rules, I am getting this error. Please would you give me some suggestions?

Reply:
suggestion: search a bit and try to understand what's going on. It's very simple. BTW you should always do your homework first and try finding the info for several hours before asking, as otherwise you won't learn (when someone tells you the answer you learn very little in practice, what's important is that you can find the info by yourself, this is how you really progress). Especially here where you need less than 5 minutes to find the answser...

Reply:
Also please use `String.format()` instead for ` throw new MacroExecutionException("The included " + includeType + " doesn't contain any heading"); `.

Reply:
I totally agree with you!. Sure, I will find the solution by searching for this error. Thanks for the motivation :).